test/clj_sqs_extended/core_test.clj

```diff
@@ -111,6 +114,61 @@
           (is (contains? stats :stopped-at))))
       (close! handler-chan))))
 
+(deftest handle-queue-terminates-after-restart-count-exceeded
+  (testing "handle-queue terminates when the restart-count exceeds the limit"
+    (let [handler-chan (chan)]
+      (fixtures/with-test-standard-queue
+        (with-redefs-fn {#'sqs/receive-message
+                         (fn [_ _ _] (HttpTimeoutException.
+                                       "Something seems to be wrong with the network."))}
```

Comment:
I would note that we're testing in the message. e.g. "Testing network failure". So that this raised exception is clearly a test in the test log and not a real network fail during testing.